homework.rb

```diff
@@ -1,6 +1,10 @@
-VARIANT = 0
+VARIANT = 8
 
 def generate
-  # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
+  letters = *("A".."Z")
+
+  letter1, letter2 = 2.times.map { letters.sample(3).join }
```

Comment:
THNQ for the detailed prompt.
As I understand, this works due to stabby lambda (lambda literal) syntax. 
In result array, lambda are just called with no args. 
TBH, I`ve rarely used that.